### DISCUSSION TITLE
profile incompatible with image

### DISCUSSION THREAD

**Discussion by stuartrexking** [libvips/libvips#3704] [Q&A]
I'm trying to convert [this](https://shopdeft.com/images/bcc911c9-0507-5357-afc3-2f21928db68c) image to webp

```
root@79fc6e75a0c5:/# wget https://shopdeft.com/images/bcc911c9-0507-5357-afc3-2f21928db68c
root@79fc6e75a0c5:/# identify bcc911c9-0507-5357-afc3-2f21928db68c 
bcc911c9-0507-5357-afc3-2f21928db68c JPEG 500x500 500x500+0+0 8-bit Gray 256c 24160B 0.000u 0:00.000
root@79fc6e75a0c5:/# cat bcc911c9-0507-5357-afc3-2f21928db68c | vips thumbnail_source --linear '[descriptor=0]' '.webp[Q=75,strip,profile=none]' 1024 > out.webp

(vips:5230): VIPS-WARNING **: 12:11:23.292: profile incompatible with image

(vips:5230): VIPS-WARNING **: 12:11:23.292: profile incompatible with image
icc_import: unable to load or find any compatible input profile
root@79fc6e75a0c5:/# ls -la out.webp 
-rw-r--r-- 1 root root 0 Oct  7 12:11 out.webp
```

I have seen [this related issue](https://github.com/libvips/libvips/issues/1314), I just can't tell how to set the profiles correctly using that command. I don't need a specific profile, I just need the resulting webp to be web friendly (so srgb?).

**Comment by jcupitt** [comment]
Hi @stuartrexking,

You have a monochrome (one band) JPEG with an sRGB profile attached to it, hence the confusion.

Your command works in libvips 8.14, perhaps you have an old libvips? There have been some improvements to the handling of mismatched profiles:

```
$ vips --version
vips-8.14.5
$ cat bcc911c9-0507-5357-afc3-2f21928db68c | vips thumbnail_source --linear '[descriptor=0]' '.webp[Q=75,strip,profile=none]' 1024 > out.webp
(vips:3147880): VIPS-WARNING **: 14:13:12.421: profile incompatible with image
```

You get a warning, but it's just a warning, and the image is generated correctly.

fwiw, I would use:

```
cat bcc911c9-0507-5357-afc3-2f21928db68c | vips thumbnail_source [descriptor=0] .webp[strip] 1024 > out.webp
```

- only use `linear` if you really have to, it's very slow
- `webpsave` is Q 75 by default
- `strip` will remove any profile
- without `linear`, there are no colour space transforms, so the mismatched profile doesn't matter

**Comment by jcupitt** [comment]
You could also write:

```
$ cat bcc911c9-0507-5357-afc3-2f21928db68c | vipsthumbnail stdin -o .webp[strip] -s 1024 > out.webp
```

The `vipsthumbnail` program has a more flexible (imagemagick-like) size specifier.

**Comment by stuartrexking** [comment]
Thanks again for the advice!

**Comment by kleisauke** [comment]
For reference, this has been fixed in libvips v8.14.3 with commit https://github.com/libvips/libvips/commit/cbe65cc060d62887ae505087a9ff49c78650dd97.
